### ISSUE TITLE
two problems encountered trying to add newline to "expand..." text

### ISSUE THREAD

**Issue by brianharvey** [jmoenig/Snap#3404]
So, sathvikrias posted this improved IF reporter in the forum:

![IF BLOCK script pic(3)](https://github.com/user-attachments/assets/b14e949d-7142-4160-8a0d-0edd306f6e87)

![untitled script pic (4)](https://github.com/user-attachments/assets/657dd982-025c-4b2b-a89a-828c8e876d50)

![untitled script pic (5)](https://github.com/user-attachments/assets/a64595ba-9d3a-486e-a974-bf78bf9b4843)

I really like that, but if you have more than one ELSE IF the block gets very wide very quickly, so I wanted to put a line break before the words "else if" so I tried "$nl else if" and got this:

![untitled script pic (3)](https://github.com/user-attachments/assets/9cba0a48-9f17-4a2e-b5f4-3edda884913d)

That's the first problem; $symbol names should end at a space and be mixable with other text.  The second problem is that, having read about some feature involving "%nl" I also tried "%nl else if."  Having done that, I am unable to click OK on the block editor.  No error message, it just silently doesn't close (but it does change the block, badly).  Even if I delete the "%nl" in the "expand..." option, that doesn't let me click OK.  All I can do is click Cancel, which leaves me with the messed up version of the block.

**Comment by brianharvey** [comment]
P.S.  I think he's right that (after fixing the nl problem) we should change the primitive to do this.

**Comment by jmoenig** [comment]
that's right, newlines currently cannot be mixed with other label elements in the expansion tokens. Known (because *I* didn't yet need it, and on my list.
I don't want to add a variadic if/else reporter just yet, I'm not convinced... but let's discuss this sometime offline (in a few weeks).

**Comment by brianharvey** [comment]
P.P.S.  I did more playing around with how the block is line wrapped and got this:

![untitled script pic (6)](https://github.com/user-attachments/assets/14a231b5-1387-4a74-8bc9-570c359f9afd)

That's pretty hideous, imho.  I get that you're treating the entire multi-line input group as a single token for purposes of wrapping the overall expression, but it took quite a while for me to see that the ELSE floating by itself at the right edge isn't vertically aligned with the text starting THEN.

I agree in principle that it's good to keep the input group together, but if so, it should be followed by a newline automatically.  And also if it needs to be broken, do it at a group boundary (so the line starting THEN would instead start with the ELSE IF that comes before it).

Happy to discuss later, but why not RESHAPE?
